neqo-transport/src/connection.rs

```diff
@@ -953,30 +936,30 @@ impl Connection {
             });
         }
 
-        let mut m = self.tls.handshake_raw(now, rec);
+        let mut m = self.crypto.tls.handshake_raw(now, rec);
 
-        if matches!(m, Ok(_)) && *self.tls.state() == HandshakeState::AuthenticationPending {
+        if matches!(m, Ok(_)) && *self.crypto.tls.state() == HandshakeState::AuthenticationPending {
```

Comment:
I think that we can remove the `matches!()` here.  The line is very long, and the state will be AuthenticationPending only if `m` matches `Ok(_)`.